src/k8s-engine/lima.ts

```diff
@@ -1285,37 +1326,42 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     const previousVersion = (await this.currentConfig)?.k3s?.version;
     const isDowngrade = previousVersion ? semver.gt(previousVersion, desiredVersion) : false;
     let commandArgs: Array<string>;
+    const enabledK3s = this.#enabledK3s = config.enabled;
 
     this.#desiredPort = config.port;
     this.setState(K8s.State.STARTING);
     this.currentAction = Action.STARTING;
     if (this.cfg?.containerEngine) {
       this.#currentContainerEngine = this.cfg.containerEngine;
     }
-
-    this.lastCommandComment = 'Starting kubernetes';
+    this.lastCommandComment = 'Starting Backend';
     await this.progressTracker.action(this.lastCommandComment, 10, async() => {
       try {
         await this.ensureArchitectureMatch();
-        if (this.progressInterval) {
-          timers.clearInterval(this.progressInterval);
+        if (enabledK3s) {
+          if (this.progressInterval) {
+            timers.clearInterval(this.progressInterval);
+          }
+          this.progressInterval = timers.setInterval(() => {
+            const statuses = [
+              this.k3sHelper.progress.checksum,
+              this.k3sHelper.progress.exe,
+              this.k3sHelper.progress.images,
+            ];
+            const sum = (key: 'current' | 'max') => {
+              return statuses.reduce((v, c) => v + c[key], 0);
+            };
+
+            this.progressTracker.numeric('Downloading Kubernetes components', sum('current'), sum('max'));
+          }, 250);
         }
-        this.progressInterval = timers.setInterval(() => {
-          const statuses = [
-            this.k3sHelper.progress.checksum,
-            this.k3sHelper.progress.exe,
-            this.k3sHelper.progress.images,
-          ];
-          const sum = (key: 'current' | 'max') => {
-            return statuses.reduce((v, c) => v + c[key], 0);
-          };
-
-          this.progressTracker.numeric('Downloading Kubernetes components', sum('current'), sum('max'));
-        }, 250);
-
-        this.lastCommandComment = 'Ensure k3 images, virtualization, check cluster configuration';
+
+        if (enabledK3s) {
+          this.lastCommandComment = 'Checking k3s images';
+          await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));
```

Comment:
Since the other items (`ensureVirtualizationSupported` / `updateConfig`) are fast, maybe do them first so the user can get faster feedback if their CPU doesn't work with virtualization? Ideally, of course, all 3 would be done in parallel.

Reply:
Bump because comments were getting lost.

Reply:
> Since the other items (`ensureVirtualizationSupported` / `updateConfig`) are fast, maybe do them first so the user can get faster feedback if their CPU doesn't work with virtualization? Ideally, of course, all 3 would be done in parallel.

`lima.ts` line 1374.
> await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));

Reply:
Done.